testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oauth/LogoutTest.java

```diff
@@ -293,7 +293,7 @@ public void successfulKLogoutAfterEmptyBackChannelUrl() throws Exception {
         }
 
         // Assert logout event triggered for backchannel logout
-        events.expectLogout(sessionId).detail(Details.REDIRECT_URI, oauth.APP_AUTH_ROOT).assertEvent();
+        events.expectLogout(sessionId).client((String) null).detail(Details.REDIRECT_URI, oauth.APP_AUTH_ROOT).assertEvent();
```

Comment:
I think we are missing a similar change (set client to event) when going through the logout endpoint. This assertion indicates we are missing the client there too when the `client_id` or `id_token_hint` is passed.

Looking at the `LogoutEndpoint#logout` (GET) method, it seems we are only setting the client when there is a mismatch between `client_id` and id_token `azp`. See https://github.com/keycloak/keycloak/blob/main/services/src/main/java/org/keycloak/protocol/oidc/endpoints/LogoutEndpoint.java.

Reply:
Looking for a better place/another place to setup the client id.

Reply:
Added the client to the event on the same place where the client gets added to the session context. Now there's even a logout error event test that reports a client id, so I it looks like a good place to set the client.